src/sentry/testutils/outbox.py

```diff
@@ -32,3 +41,25 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
     with TaskRunner(), assume_test_silo_mode(SiloMode.MONOLITH):
         while enqueue_outbox_jobs():
             pass
+
+
+def assert_webhook_outboxes(
+    factory_request: WSGIRequest,
+    webhook_identifier: WebhookProviderIdentifier,
+    region_names: List[str],
+):
+    expected_payload = ControlOutbox.get_webhook_payload_from_request(request=factory_request)
+    expected_payload_dict = dataclasses.asdict(expected_payload)
+    assert ControlOutbox.objects.count() == len(region_names)
+    region_names_set = set(region_names)
+    for cob in ControlOutbox.objects.all():
+        assert cob.payload == expected_payload_dict
+        assert cob.shard_scope == OutboxScope.WEBHOOK_SCOPE
+        assert cob.shard_identifier == webhook_identifier
+        assert cob.category == OutboxCategory.WEBHOOK_PROXY
+        try:
+            region_names_set.remove(cob.region_name)
+        except KeyError:
+            raise Exception(
+                f"Found outbox for '{cob.region_name}', which was not in region_names: {str(region_names_set)}"
+            )
```

Comment:
@leeandher for completeness, it might be good to also assert that `region_names_set` is empty after the for loop. This should give us confidence that all the region names are exhausted.